application/server/server.go

```diff
@@ -547,21 +547,28 @@ func textDocumentDidSaveHandler() jrpc2.Handler {
 		bgCtx := context.Background()
 		c := config.CurrentConfig()
 		logger := c.Logger().With().Str("method", "TextDocumentDidSaveHandler").Logger()
-
 		logger.Info().Interface("params", params).Msg("Receiving")
+
+		autoScanEnabled := c.IsAutoScanEnabled()
+
 		di.FileWatcher().SetFileAsSaved(params.TextDocument.URI)
 		filePath := uri.PathFromUri(params.TextDocument.URI)
 
-		f := workspace.Get().GetFolderContaining(filePath)
-		autoScanEnabled := config.CurrentConfig().IsAutoScanEnabled()
-		if f != nil && autoScanEnabled {
-			go f.ScanFile(bgCtx, filePath)
-		} else {
+		ws := workspace.Get()
+		if ws != nil && autoScanEnabled && uri.IsDotSnykFile(params.TextDocument.URI) {
+			go ws.ScanWorkspace(bgCtx)
+			return nil, nil
+		}
+
+		f := ws.GetFolderContaining(filePath)
+		if f != nil {
 			if autoScanEnabled {
-				logger.Warn().Str("documentURI", filePath).Msg("Not scanning, file not part of workspace")
+				go f.ScanFile(bgCtx, filePath)
```

Comment:
This may cause a double scan, right? First the scan above and afterwards the f.ScanFile. I'd think we can combine it :)